azure-devops/azext_devops/test/team/test_extension.py

```diff
@@ -0,0 +1,157 @@
+        self.get_credential_patcher.stop()
+        
+
+    def test_list_extensions(self):
+        list_extensions('false','false',self._TEST_DEVOPS_ORGANIZATION, 'off')
+                
+        #assert
+        self.mock_get_installed_extensions.assert_called_once_with(include_disabled_extensions=False)
+
+    def test_list_extensions_include_disabled_extension(self):
+        list_extensions('false','true',self._TEST_DEVOPS_ORGANIZATION, 'off')
+                
+        #assert
+        self.mock_get_installed_extensions.assert_called_once_with(include_disabled_extensions=True)
+
+    def test_list_extension_include_builtin_extension(self):
+        extensions = []
+        extensions.append(MockInstalledExtension('builtIn, installed'))
+        extensions.append(MockInstalledExtension('builtIn, installed'))
+        extensions.append(MockInstalledExtension('installed'))
+
+        self.mock_get_installed_extensions.return_value = extensions
+
+        result = list_extensions('true','true',self._TEST_DEVOPS_ORGANIZATION, 'off')
+
+        self.assertEqual(len(result), 3)
+
+    def test_list_extension_not_include_builtin_extension(self):
+        extensions = []
+        extensions.append(MockInstalledExtension('builtIn, installed'))
+        extensions.append(MockInstalledExtension('builtIn, installed'))
+        extensions.append(MockInstalledExtension('installed'))
+
+        self.mock_get_installed_extensions.return_value = extensions
+
+        result = list_extensions('false','true',self._TEST_DEVOPS_ORGANIZATION, 'off')
+
+        self.assertEqual(len(result), 1)
+
+    def test_get_extension(self):
+        get_extension('ms', 'code-search', self._TEST_DEVOPS_ORGANIZATION, 'off')
+
+        self.mock_get_installed_extension.assert_called_once_with(publisher_name='ms', extension_name='code-search')
+
+    def test_install_extension(self):
+        install_extension('ms', 'code-search', self._TEST_DEVOPS_ORGANIZATION, 'off')
+
+        self.mock_install_extension.assert_called_once_with(publisher_name='ms', extension_name='code-search')
+        
+    def test_uninstall_extension(self):
+        uninstall_extension('ms', 'code-search', self._TEST_DEVOPS_ORGANIZATION, 'off')
+
+        self.mock_uninstall_extension.assert_called_once_with(publisher_name='ms', extension_name='code-search')
+
+    def test_enable_extension(self):
+        extension = MockInstalledExtension('builtIn', MockInstalledExtension('disabled , buildIn , multiVersion'))
+        self.mock_get_installed_extension.return_value = extension
+
+        enable_extension('ms', 'code-search', self._TEST_DEVOPS_ORGANIZATION, 'off')
+
+        self.mock_update_installed_extension.assert_called_once()
+        udpate_extension_object = self.mock_update_installed_extension.call_args_list[0][0]
+        install_state_flags = udpate_extension_object[0].install_state.flags
+        # self.assertTrue('disabled' not in install_state_flags)
```

Comment:
nit: Should be uncommented?

Reply:
not required we have better check in next line